File: user-api/ts-sdk/src/sdk/models/operations/activateuser.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class ActivateUserQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=token" })
  token: string;
}


export class ActivateUserRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: ActivateUserQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.UserActivationPayload;
}


export class ActivateUserResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: user-api/go-sdk/pkg/models/operations/activateuser.go
package operations

import (
	"openapi/pkg/models/shared"
)

type ActivateUserQueryParams struct {
	Token string `queryParam:"style=form,explode=true,name=token"`
}

type ActivateUserRequest struct {
	QueryParams ActivateUserQueryParams
	Request     *shared.UserActivationPayload `request:"mediaType=application/json"`
}

type ActivateUserResponse struct {
	ContentType string
	StatusCode  int64
}
